### PR TITLE
OIDC Discovery Provider: Support listening on a unix socket

### PR DESCRIPTION
**Pull request by mcpherrinm**
**Pull Request check list**

- [ 👍] Commit conforms to CONTRIBUTING.md?
- [ 👍] Proper tests/regressions included?
- [ 👍] Documentation updated?

**Affected functionality**
New feature for OIDC Discovery Provider: support listening on unix socket

**Description of change**
This adds a new, documented option as an alternative to listening with ACME certificates.

I plan to use this as part of our service mesh behind an ingress proxy.  The standard interface in our deployment is to use a unix socket, as we do not directly expose applications externally.

### CODE REVIEW

**Comment by mcpherrinm** [comment]
Force pushed for missing signoff.

File: support/oidc-discovery-provider/config.go [hunk 1]
```diff
@@ -130,11 +134,16 @@ func ParseConfig(hclConfig string) (_ *Config, err error) {
 
 	switch {
 	case c.ACME == nil:
-		if c.InsecureAddr == "" {
+		if c.InsecureAddr == "" && c.ListenSocketPath == "" {
 			return nil, errs.New("acme section must be configured")
```
Comment (azdagron) [inline support/oidc-discovery-provider/config.go]: "either acme or listen_socket_path must be configured"

File: support/oidc-discovery-provider/README.md [hunk 2]
```diff
@@ -30,16 +30,19 @@ The provider has the following command line flags:
...
-| `domain`           | string  | required | The domain the provider is being served from | |
-| `log_format`       | string  | optional | Format of the logs (either `"TEXT"` or `"JSON"`) | `""` |
-| `log_level`        | string  | required | Log level (one of `"error"`,`"warn"`,`"info"`,`"debug"`) | `"info"` |
-| `log_path`         | string  | optional | Path on disk to write the log | |
-| `log_requests`     | bool    | optional | If true, all HTTP requests are logged at the debug level | false |
-| `registration_api` | section | required\* | Provides Registration API details. Required unless the `workload_api` section is defined | |
-| `workload_api`     | section | required\* | Provides Workload API details. Required unless the `registration_api` section is defined | |
+| Key                  | Type    | Required? | Description                              | Default |
+| -------------------- | --------| ---------| ----------------------------------------- | ------- |
+| `acme`               | section | optional | Provides the ACME configuration. | |
```
Comment (azdagron) [inline support/oidc-discovery-provider/README.md]: nit: maybe put `required*` on both acme and listen_socket_path and annotate your note below with the asterisk? If I was just scanning that list to get a feel for what I needed to configure, I might not notice the note below. The `*` might draw the eyes down to the qualifying note.

File: support/oidc-discovery-provider/config.go (hunk 1 above)
Comment (mcpherrinm) [reply support/oidc-discovery-provider/config.go]: Updated

File: support/oidc-discovery-provider/README.md (hunk 2 above)
Comment (mcpherrinm) [reply support/oidc-discovery-provider/README.md]: Github flavored markdown doesn't support footnotes, but I've updated this with [1] style numbered-footnotes to indicate pairs where one or the other option must be set.

**Comment by mcpherrinm** [comment]
fixed test

**Comment by mcpherrinm** [comment]
@azdagron good for merge?

**Pull request merged**